forms-flow-api/src/formsflow_api/services/authorization.py

```diff
@@ -3,19 +3,203 @@
 import datetime
 from typing import Dict, List
 
+from flask import current_app
 from formsflow_api_utils.exceptions import BusinessException
 from formsflow_api_utils.utils.user_context import UserContext, user_context
+from sqlalchemy.orm.attributes import flag_modified
 
 from formsflow_api.constants import BusinessErrorCode
-from formsflow_api.models import Application, Authorization, AuthType
+from formsflow_api.models import Application, Authorization, AuthType, db
 from formsflow_api.schemas import ApplicationSchema
+from formsflow_api.services.external.analytics_api import RedashAPIService
 
 application_schema = ApplicationSchema()
 
 
 class AuthorizationService:
     """This class manages authorization service."""
 
+    def check_and_update_name_mismatch(  # pylint: disable=too-many-arguments, too-many-positional-arguments
+        self,
+        auth_name: str,
+        dashboard_id: str,
+        dashboard: Dict[str, str],
+        auth_record: Authorization,
+        updates_needed: List[Dict],
+        result: List[Dict],
+    ):
+        """Check if the name in auth record matches the dashboard name and update if needed."""
+        if auth_name != dashboard["name"]:
+            current_app.logger.info(
+                f"Name mismatch detected for {dashboard_id}: "
+                f"Auth has '{auth_name}', Dashboard has '{dashboard['name']}'"
+            )
+            updates_needed.append(
+                {
+                    "resource_id": dashboard_id,
+                    "new_name": dashboard["name"],
+                    "auth_object": auth_record,
+                }
+            )
+            auth_record.resource_details["name"] = dashboard["name"]
+        # Always add to result (even if no update needed)
+        result.append(self._as_dict(auth_record))
+        return updates_needed, result
+
+    def update_auth_records(self, auth_records):
+        """Update dashboard authorization records."""
+        current_app.logger.info("Updating auth records in the database")
+        for auth_record in auth_records:
+            current_app.logger.info(
+                f"Updating auth record: {auth_record['resource_id']} with new name: {auth_record['new_name']}"
+            )
+            auth_object = auth_record["auth_object"]
+            auth_object.resource_details = {"name": auth_record["new_name"]}
```

Comment:
Aren't we doing same action in L#44? Do we need this here or will the `flag_modified(...)` statement below be enough?

Reply:
Yes. The flag_modified(...) statement itself is enough.  Updated code